docs/icicle/primitives/poseidon.md

```diff
@@ -0,0 +1,233 @@
+# Poseidon
+
+Poseidon hash is a popular hash in the ZK ecosystem primarily because its optimized to work over large prime fields, a common setting for ZK proofs, thereby minimizing the number of multiplicative operations required.
```

Comment:
```suggestion
Poseidon hash is a popular hashing algorithm in the ZK ecosystem primarily because it is optimized to work over large prime fields, a common setting for ZK proofs, thereby minimizing the number of multiplicative operations required.
```